enterprise/frontend/src/embedding/data-picker/DataSelector/DataSelector.jsx

```diff
@@ -340,36 +307,23 @@ export class UnconnectedDataSelector extends Component {
       selectedSchema.database.id !== SAVED_QUESTIONS_VIRTUAL_DB_ID;
 
     const onStepMissingSchemaAndTable =
-      !selectedSchema &&
-      !selectedTable &&
-      (activeStep === TABLE_STEP || activeStep === FIELD_STEP);
-
-    const onStepMissingTable = !selectedTable && activeStep === FIELD_STEP;
+      !selectedSchema && !selectedTable && activeStep === TABLE_STEP;
 
     const invalidTable =
       selectedSchema &&
       selectedTable &&
       !isVirtualCardId(selectedTable.id) &&
       selectedTable.schema.id !== selectedSchema.id;
 
-    const invalidField =
```

Comment:
the whole thing can be removed because `selectedField` is falsy that makes `invalidField` false. Then we could remove block on line 371 (previous diff). After that this variable isn't used anymore